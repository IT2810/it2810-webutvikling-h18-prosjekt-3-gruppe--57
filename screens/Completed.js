import React from 'react';
import {
    ScrollView,
    Text,
    View,
    TouchableHighlight,
    Image,
    Modal,
    Button,
} from 'react-native';
import {LinearGradient, MapView} from 'expo';
import { Marker } from "react-native-maps";
import Storage from "../components/Storage.js";
import createStyles from "../styles/CompletedStyle.js";
import ActionButton from "react-native-action-button";
import color from "../constants/Colors";

const styles = createStyles();

export default class Completed extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            reminders: [],
            modalVisible: false,
            toggleMap: function () {
                this.setState({ modalVisible: false });
            },
        };
    }

    static navigationOptions = {
        header: null,
    };

    async componentDidMount() {
        //refresh list when component is focused, necessary when exiting modal
        this.props.navigation.addListener("willFocus", ()=>{
            Storage.getCompletedRemindersSorted().then((res)=>{
                this.setState({reminders:res});
            });
        });
    }

    render() {
        return (
            <View style={{flex: 1, backgroundColor: '#f3f3f3', borderRadius: 0}}>
                <Modal
                    animationType="slide"
                    transparent={false}
                    visible={this.state.modalVisible}
                    onRequestClose={this.state.toggleMap.bind(this)}
                    >
                    <MapView
                        style={{ flex: 1, width: 400, height: 600 }}
                        region={this.state.reminders.length > 0 ? this.state.reminders[0].location : { latitude: 63.413, longitude: 10.406, latitudeDelta: 0, longitudeDelta: 0.4}}
                    >
                        {
                            this.state.reminders.map(reminder => (
                                reminder.location ? 
                                <Marker
                                    key={reminder.id}
                                    title={reminder.reminder}
                                    description={reminder.date}
                                    coordinate={reminder.location}
                                /> : <View key={reminder.id}/>
                            ))
                        }
                        {/*<Button onClick={this.setState({modalVisible: false})}>Exit</Button> */}
                    </MapView>
                </Modal>
                <ScrollView style={styles.container}>
                    {
                        this.state.reminders.length > 0 ? this.state.reminders.map((l, i) => (
                                <View style={styles.item} key={i}>
                                    <View style={styles.shadow}>
                                        <TouchableHighlight underlayColor={"#f1f1f1"} style={{borderRadius: 10}}
                                        >
                                            <LinearGradient
                                                colors={l.locked ? color.colorPalletError : color.colorPalletGreen}
                                                style={styles.gradient}>
                                                <View style={styles.info1}>
                                                    <Text
                                                        style={l.locked ? styles.reminderTextLocked : styles.reminderText}>
                                                        {l.reminder}
                                                    </Text>
                                                    <Text style={l.locked ? styles.dateTextLocked : styles.dateText}>
                                                        {l.date}
                                                    </Text>
                                                </View>
                                            </LinearGradient>
                                        </TouchableHighlight>
                                    </View>
                                </View>
                            )) :
                            <View style={styles.emptyTextView}>
                                <Image
                                    source={{uri: 'https://thumbs.gfycat.com/AffectionateDimIberianlynx-small.gif'}}
                                    style={{height: 200, width: 200}}/>
                                <Text style={styles.emptyText}>There's nothing here</Text>
                            </View>
                    }
                </ScrollView>
                <ActionButton buttonColor="#000" title="New Reminder"
                    onPress={() => this.setState({modalVisible:true})}>
                </ActionButton>
            </View>
        );
    }
}