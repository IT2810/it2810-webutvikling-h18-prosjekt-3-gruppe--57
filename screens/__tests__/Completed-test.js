import React from "react";
import "react-native";
import renderer from "react-test-renderer";
import Completed from "../Completed";
import ShallowRenderer from 'react-test-renderer/shallow';


it('Snapshot of shallowRender.', ()=> {
    const renderer = new ShallowRenderer();
    renderer.render(<Completed/>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});
