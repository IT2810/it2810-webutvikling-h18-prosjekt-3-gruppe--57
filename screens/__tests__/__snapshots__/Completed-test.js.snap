// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of shallowRender. 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#f3f3f3",
      "borderRadius": 0,
      "flex": 1,
    }
  }
>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <MapView
      region={
        Object {
          "latitude": 63.413,
          "latitudeDelta": 0,
          "longitude": 10.406,
          "longitudeDelta": 0.4,
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 600,
          "width": 400,
        }
      }
    />
    <Button
      onPress={[Function]}
      title="Go back"
    />
  </Component>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 0,
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 0,
          "flex": 1,
          "height": 1184,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://thumbs.gfycat.com/AffectionateDimIberianlynx-small.gif",
          }
        }
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#646464",
            "fontSize": 17,
            "textAlign": "center",
          }
        }
      >
        There's nothing here
      </Text>
    </Component>
  </ScrollViewMock>
  <ActionButton
    active={false}
    activeOpacity={0.85}
    autoInactive={true}
    backdrop={false}
    backgroundTappable={false}
    bgColor="transparent"
    bgOpacity={1}
    buttonColor="#000"
    buttonText="+"
    buttonTextStyle={Object {}}
    degrees={45}
    fixNativeFeedbackRadius={false}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    offsetX={30}
    offsetY={30}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOn={[Function]}
    outRangeScale={1}
    position="right"
    resetToken={null}
    size={56}
    spacing={20}
    title="New Reminder"
    useNativeFeedback={true}
    verticalOrientation="up"
  />
</Component>
`;
